### PR TITLE
EVG-13162 add functions to remove/add annotation issues

### PR DESCRIPTION
**Pull request by ablack12**

### CODE REVIEW

**Comment by ablack12** [comment]
I threw in the "move issue from here to there" bit cause it was pretty related.

File: model/annotations/task_annotations_test.go [hunk 1]
```diff
@@ -38,8 +38,118 @@ func TestGetLatestExecutions(t *testing.T) {
...
+	assert.NotNil(t, annotationFromDB)
+	assert.Len(t, annotationFromDB.SuspectedIssues, 1)
+	assert.Equal(t, "not.annie.black", annotationFromDB.SuspectedIssues[0].Source.Author)
+}
+
+func TestMoveIssueToSuspectedIssue(t *testing.T) {
+	issue1 := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &Source{Author: "this will be overridden"}}
+	issue2 := IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &Source{Author: "evergreen user"}}
+	assert.NoError(t, db.Clear(Collection))
+	a := TaskAnnotation{TaskId: "t1", Issues: []IssueLink{issue1, issue2}, SuspectedIssues: []IssueLink{issue2}}
```
Comment (malikchaya2) [inline model/annotations/task_annotations_test.go:125]: We shouldn't ever have duplicates between these two fields, so having issue2 in both lists is probably not the best setup.

**Review by malikchaya2 — CHANGES_REQUESTED** [review]

File: model/annotations/task_annotations_test.go (hunk 1 above)
Comment (ablack12) [reply model/annotations/task_annotations_test.go:125]: Just laziness haha but I respect that

**Review by malikchaya2 — APPROVED** [review]

File: model/annotations/task_annotations.go [hunk 2]
```diff
@@ -52,3 +63,77 @@ func GetLatestExecutions(annotations []TaskAnnotation) []TaskAnnotation {
...
+	}
+
+	_, err := db.Upsert(
+		Collection,
+		ByTaskIdAndExecution(taskId, execution),
+		bson.M{
+			"$push": bson.M{SuspectedIssuesKey: issue},
+		},
+	)
+	return errors.Wrapf(err, "problem adding task annotation suspected issue for task '%s'", taskId)
```
Comment (malikchaya2) [inline model/annotations/task_annotations.go:122]: Will this return an error even when the error is nil?

Comment (ablack12) [reply model/annotations/task_annotations.go:122]: Nope!

**Pull request merged**